eq-author-api/schema/resolvers/pages/index.js

```diff
@@ -11,42 +20,77 @@ const Resolvers = {};
 
 Resolvers.Page = {
   __resolveType: ({ pageType }) => pageType,
-  position: ({ id }, args, ctx) => {
-    const section = getSectionByPageId(ctx, id);
-    return findIndex(section.pages, { id });
-  },
+  position: ({ id }, args, ctx) => findIndex(getPagesBySectionId(ctx, id), id),
 };
 
 Resolvers.Mutation = {
   movePage: createMutation((_, { input }, ctx) => {
-    const section = getSectionByPageId(ctx, input.id);
-    const removedPage = first(remove(section.pages, { id: input.id }));
-    if (input.sectionId === section.id) {
-      section.pages.splice(input.position, 0, removedPage);
+    const { id: pageId, sectionId, folderId, position } = input;
+
+    const page = getPageById(ctx, pageId);
+    const oldSection = getSectionByPageId(ctx, pageId);
+    const newSection = getSectionById(ctx, sectionId);
+
+    oldSection.folders.map(folder => {
+      const { enabled, pages } = folder;
+      const index = pages.findIndex(({ id }) => id === page.id);
+
+      if (index > -1) {
+        pages.splice(index, 1);
+
+        if (enabled && !pages.length) {
+          pages.push(createQuestionPage());
+        }
+      }
+    });
+
+    oldSection.folders = filter(
+      oldSection.folders,
+      ({ pages }) => pages.length
+    );
+
+    if (folderId) {
+      const folder = getFolderById(ctx, folderId);
+      folder.pages.splice(position, 0, page);
+      page.folderId = folder.id;
     } else {
-      const newsection = find(ctx.questionnaire.sections, {
-        id: input.sectionId,
-      });
-      newsection.pages.splice(input.position, 0, removedPage);
+      const { folders } = newSection;
+      const newFolder = createFolder({ pages: [page] });
+      folders.splice(position, 0, newFolder);
+      page.folderId = newFolder.id;
     }
-    return removedPage;
+
+    return { ...page, sectionId: newSection.id };
   }),
+
+  // still needs testing
   deletePage: createMutation((_, { input }, ctx) => {
     const section = getSectionByPageId(ctx, input.id);
-    const removedPage = first(remove(section.pages, { id: input.id }));
-    onPageDeleted(ctx, section, removedPage);
+    const { previous } = getMovePosition(section, input.id, 0);
+    section.folders[previous.folderIndex].pages.splice(previous.pageIndex, 1);
+
+    onPageDeleted(ctx, section, previous.page);
+
+    if (!section.folders[previous.folderIndex].pages.length) {
+      // section.folders.splice(previous.folderIndex, 1);
+    }
```

Comment:
```suggestion
```

Reply:
Only reason i've left this in is because I'm unsure whether I need to remove a folder if there is no pages left in it. Or if that action will be completed on the front-end

Reply:
Iirc we're going to treat Folders the same way we do Sections in that regard. That is, if there are no pages left within the folder, we create a new blank page within it.